### PR TITLE
fix(protocol-designer): analytics opt in modal fixes

### PR DESCRIPTION
**Pull request by jerader**
closes AUTH-1142

# Overview

Fix when the opt in modal shows up and also adjust the copy and CTAs.

## Test Plan and Hands on Testing

Test that this works in your dev env with the gate modal ff turned on like so: `make -C protocol-designer dev OT_PD_SHOW_GATE=true`. Test that if you select to accept and then go to the settings to toggle it off, it doesn't show up again unless you clear local cache.

## Changelog

update modal to match designs

code wise, i added an appVersion prop to the opt_in action. If the appVersion is undefined, it will trigger the modal. also if the opt_in is null, it will trigger the modal. i believe this should force the user to see the modal even if they previously selected false.

## Risk assessment

low

### CODE REVIEW

File: protocol-designer/src/ProtocolRoutes.tsx [hunk 1]
```diff
@@ -59,8 +59,6 @@ export function ProtocolRoutes(): JSX.Element {
     path: '/',
   }
   const allRoutes: RouteProps[] = [...pdRoutes, landingPage]
-  const showGateModal =
-    process.env.NODE_ENV === 'production' || process.env.OT_PD_SHOW_GATE
```
Comment (jerader) [inline protocol-designer/src/ProtocolRoutes.tsx:63]: i don't think this is necessarily needed? i think its ok if we run into in our dev envs/non-production builds

File: protocol-designer/src/analytics/middleware.ts
Comment (koji) [inline protocol-designer/src/analytics/middleware.ts:1]: this is not directly related to this pr but seems that `dispatch` in `trackEventMiddleware` isn't used.

File: protocol-designer/src/analytics/middleware.ts [hunk 2]
```diff
@@ -499,7 +499,7 @@ export const trackEventMiddleware: Middleware<BaseState, any> = ({
   // NOTE: this is the Redux state AFTER the action has been fully dispatched
   const state = getState()
 
-  const optedIn = getHasOptedIn(state as BaseState) ?? false
+  const optedIn = getHasOptedIn(state as BaseState)?.hasOptedIn ?? false
```
Comment (koji) [inline protocol-designer/src/analytics/middleware.ts:502]: not 100% sure but the request is mixpanel is `on` as the default setting so this might be `true`?

File: protocol-designer/src/analytics/mixpanel.ts
Comment (koji) [inline protocol-designer/src/analytics/mixpanel.ts:1]: Jamey adds the error handling to mixpanel initialization so probably it would be good to align with that.
https://github.com/Opentrons/opentrons/pull/17104/files


like this
```ts
import mixpanel from 'mixpanel-browser'
import { getIsProduction } from '../networking/opentronsWebApi'
import { getHasOptedIn } from './selectors'
import type { BaseState } from '../types'

export type AnalyticsEvent =
  | {
      name: string
      properties: Record<string, unknown>
      superProperties?: Record<string, unknown>
    }
  | { superProperties: Record<string, unknown> }

const MIXPANEL_ID = getIsProduction()
  ? process.env.OT_PD_MIXPANEL_ID
  : process.env.OT_PD_MIXPANEL_DEV_ID

const MIXPANEL_OPTS = {
  opt_out_tracking_by_default: true,
}

export function initializeMixpanel(state: BaseState): void {
  const optedIn = getHasOptedIn(state)?.hasOptedIn ?? false
  if (MIXPANEL_ID != null) {
    try {
      console.debug('Initializing Mixpanel', { optedIn })
      mixpanel.init(MIXPANEL_ID, MIXPANEL_OPTS)
      setMixpanelTracking(optedIn)
      trackEvent({ name: 'appOpen', properties: {} }, optedIn)
    } catch (error) {
      console.error('Error initializing Mixpanel:', error)
    }
  } else {
    console.warn('MIXPANEL_ID not found; this is a bug if build is production')
  }
}

export function trackEvent(event: AnalyticsEvent, optedIn: boolean): void {
  console.debug('Trackable event', { event, optedIn })
  if (MIXPANEL_ID != null && optedIn) {
    try {
      if ('superProperties' in event && event.superProperties != null) {
        mixpanel.register(event.superProperties)
      }
      if ('name' in event && event.name != null) {
        mixpanel.track(event.name, event.properties)
      }
    } catch (error) {
      console.error('Error tracking event:', error)
    }
  }
}

export function setMixpanelTracking(optedIn: boolean): void {
  if (MIXPANEL_ID != null) {
    try {
      if (optedIn) {
        console.debug('User has opted into analytics; tracking with Mixpanel')
        mixpanel.opt_in_tracking()
        mixpanel.register({
          appVersion: process.env.OT_PD_VERSION,
          appName: 'protocolDesigner',
        })
      } else {
        console.debug(
          'User has opted out of analytics; stopping Mixpanel tracking'
        )
        mixpanel.opt_out_tracking()
        mixpanel.reset()
      }
    } catch (error) {
      console.error('Error setting Mixpanel tracking:', error)
    }
  }
}

```

Comment (jerader) [reply protocol-designer/src/analytics/mixpanel.ts:1]: thanks Koji!

File: protocol-designer/src/components/SettingsPage/SettingsApp.tsx [hunk 3]
```diff
@@ -72,8 +69,14 @@ export function SettingsApp(): JSX.Element {
...
               <ToggleButton
                 className={styles.toggle_button}
-                toggledOn={Boolean(hasOptedIn)}
-                onClick={() => dispatch(_toggleOptedIn())}
+                toggledOn={Boolean(analytics.hasOptedIn)}
+                onClick={() =>
+                  dispatch(
+                    analytics.hasOptedIn
+                      ? analyticsActions.optOut
+                      : analyticsActions.optIn
```
Comment (koji) [inline protocol-designer/src/components/SettingsPage/SettingsApp.tsx]:
```suggestion
                    analytics.hasOptedIn
                      ? analyticsActions.optOut()
                      : analyticsActions.optIn()
```

**Comment by jerader** [comment]
changed this back to a draft because the modal isn't working as I thought it was

File: protocol-designer/src/components/SettingsPage/SettingsApp.tsx [hunk 4]
```diff
@@ -20,13 +20,10 @@ import { FeatureFlagCard } from './FeatureFlagCard/FeatureFlagCard'
 
 export function SettingsApp(): JSX.Element {
   const dispatch = useDispatch()
-  const hasOptedIn = useSelector(analyticsSelectors.getHasOptedIn)
+  const analytics = useSelector(analyticsSelectors.getHasOptedIn)
```
Comment (koji) [inline protocol-designer/src/components/SettingsPage/SettingsApp.tsx]:
```suggestion
  const { hasOptedIn } = useSelector(analyticsSelectors.getHasOptedIn)
```

File: protocol-designer/src/organisms/GateModal/index.tsx [hunk 5]
```diff
@@ -22,10 +21,10 @@ const EULA_URL = 'https://opentrons.com/eula'
 
 export function GateModal(): JSX.Element | null {
   const { t } = useTranslation('shared')
-  const hasOptedIn = useSelector(analyticsSelectors.getHasOptedIn)
+  const analytics = useSelector(analyticsSelectors.getHasOptedIn)
```
Comment (koji) [inline protocol-designer/src/organisms/GateModal/index.tsx]:
```suggestion
  const {appVersion, hasOptedIn } = useSelector(analyticsSelectors.getHasOptedIn)
```

File: protocol-designer/src/pages/Settings/index.tsx [hunk 6]
```diff
@@ -31,6 +31,7 @@ import { BUTTON_LINK_STYLE } from '../../atoms'
 import { actions as featureFlagActions } from '../../feature-flags'
 import { getFeatureFlagData } from '../../feature-flags/selectors'
 import type { FlagTypes } from '../../feature-flags'
+import { OLDEST_MIGRATEABLE_VERSION } from '../../load-file/migration'
```
Comment (koji) [inline protocol-designer/src/pages/Settings/index.tsx]:
```suggestion
```

**Comment by koji** [comment]
> changed this back to a draft because the modal isn't working as I thought it was

i guess that might be dispatch part that I pointed out.

File: protocol-designer/src/analytics/middleware.ts
Comment (jerader) [reply protocol-designer/src/analytics/middleware.ts:1]: i'll leave that to a follow up unless you think its affecting the modal?

File: protocol-designer/src/ProtocolRoutes.tsx [hunk 7]
```diff
@@ -59,9 +59,6 @@ export function ProtocolRoutes(): JSX.Element {
...
-
```
Comment (jerader) [inline protocol-designer/src/ProtocolRoutes.tsx:64]: is it cool with yall if i remove this? I don't think it'll get too annoying throughout pd development and it didn't seem to be working correctly in production (i tested my PD in prod and even with optIn as undefined, the modal still didn't show up) What do you think @shlokamin @koji @ncdiehl11 ?

**Comment by koji** [comment]
design
Screenshot 2024-12-16 at 10 08 23 AM

sandbox
Screenshot 2024-12-16 at 10 08 28 AM

from the design, need to remove the header part

**Comment by koji** [comment]
I think there is another issue.
when displaying `GetModal`,  all buttons are not clickable and there is no other option but to press confirm.

File: protocol-designer/src/analytics/middleware.ts (hunk 2 above)
Comment (jerader) [reply protocol-designer/src/analytics/middleware.ts:502]: i think it is ok if it is off or false until the user presses the button in the modal. Since its the only button available to press in the modal, then the user has to turn it on.

**Comment by jerader** [comment]
> I think there is another issue. when displaying `GetModal`, all buttons are not clickable and there is no other option but to press confirm.

Isn't that the intended behavior though? @koji

**Comment by koji** [comment]
> > I think there is another issue. when displaying `GetModal`, all buttons are not clickable and there is no other option but to press confirm.
> 
> Isn't that the intended behavior though? @koji

okay, i thought the desktop still allows to turn off the mixpanel setting when installing, but now without clicking agree, Opentrons app isn't installed.

The behavior is aligned with the Desktop app.

**Comment by koji** [comment]
need to add the following to .eslintignore

```
**/.yarn-cache/**
```

**Review by koji — APPROVED** [review]
lgtm
